cdap-ui/app/features/hydratorplusplus/controllers/list-ctrl.js

```diff
@@ -214,6 +215,7 @@ angular.module(PKG.name + '.feature.hydratorplusplus')
         })
         .then(
           function success() {
+            myLoadingService.hideLoadingIconImmediate();
```

Comment:
I believe we could chain all the events one after the other since we are using promises. Something like this,

```
  myLoadingService.showLoadingIcon()
    .then(() => {
        return mySettings.get('hydratorDrafts');
     })
     .then((res) => {
      ...
     })
     .then(myLoadingService.hideLoadingIconImmediate)
     .then(() => {
        myAlertOnValium.show({...});
     })
    .catch((err) => { ... })
```